### ISSUE TITLE
Incorrectly extracted chemistry information.

### ISSUE THREAD

**Issue by haydn-jones** [grobidOrg/grobid#1249]
I have a *large* number of PDFs which have a series of sections/subsections that look like this:

![Image](https://github.com/user-attachments/assets/57659b59-2ff5-4cfe-ad71-ed02a80c831e)

After processing all of them, I've realized that these sections are often either being identified as figures or even as bib references, which is causing a lot of issues with my pipeline. I've also seen other various issues with documents like these (missing paragraph breaks and such), but they are largely not as problematic as these sections getting lost.

Heres an example of *part* of one of these sections getting turned into a bib ref
```xml
<biblStruct xml:id="b0">
	<monogr>
		<title level="m" type="main">Orange solid, yield: 73%. 1 H NMR (400 MHz</title>
		<idno>1.7.1. 2-(3-Bromopropyl)-6-(methylamino)-1H-benzo[de]iso- quinoline-1</idno>
		<imprint>
			<biblScope unit="volume">3</biblScope>
		</imprint>
	</monogr>
	<note>H)-dione (7a) CDCl 3 ): d 8.62 (d, J = 6.8 Hz, 1H), 8.53 (d, J = 8.4 Hz, 1H), 8.13 (d, J = 9.2 Hz, 1H), 7.67 (t, J = 8.4 Hz, 1H</note>
</biblStruct>
```

And from the same PDF, several of them were turned into a figure:
```xml
<figure xmlns="http://www.tei-c.org/ns/1.0" xml:id="fig_7"><head>5.
					1 .</head><label>1</label><figDesc>Synthesis 5.1.1. General
					procedure for the preparation of 1a-1d6-Bromobenzo[de]-isochromene-1,3-dione
					(1.94 g, 7.00 mmol) was dissolved in 20 ml ethanol. Then corresponding primary
					amine (7.70 mmol) was added, and the mixture was stirred at 60 °C for 5-6 h. The
					mixture was cooled to room temperature and evaporated in vacuum to obtain the
					residue. Then the residue was purified on silica gel chromatography (PE:EA =
					10:1, V/V) to provide 1a-1d.5.1.1.1.
					6-Bromo-2-methyl-1H-benzo[de]isoquinoline-1,3(2H)dione (1a).White solid, yield:
					90%.<ref type="bibr" target="#b11">1</ref> H NMR (400 MHz, CDCl 3 ): d 8.68 (d,
					J = 7.2 Hz, 1H), 8.59 (d, J = 8.4 Hz, 1H), 8.44 (d, J = 8.0 Hz, 1H), 8.06 (d, J
					= 7.6 Hz, 1H), 7.86 (t, J = 8.4 Hz, 1H), 3.57 (s, 3H); MS(ESI) calcd for C 13 H
					9 BrNO 2 [M+H] + 289.0, found: 289.0. 5.1.1.2.
					6-Bromo-2-butyl-1H-benzo[de]isoquinoline-1,3(2H)dione (1b). White solid, yield:
					80%. 1 H NMR (400 MHz, CDCl 3 ) d 8.66 (d, J = 7.2 Hz, 1H), 8.56 (d, J = 8.8 Hz,
					1H), 8.41 (d, J = 8.0 Hz, 1H), 8.04 (d, J = 8.0 Hz, 1H), 7.85 (t, J = 8.0 Hz,
					1H), 4.19 (t, J = 7.6 Hz, 1H), 1.77-1.69 (m, 2H), 1.51-1.42 (m, 2H), 1.00 (t, J
					= 7.2 Hz, 3H); MS(ESI) calcd for C 16 H 15 BrNO 2 [M+H] + 331.0, found: 331.0.
					5.1.1.3. 6-Bromo-2-octyl-1H-benzo[de]isoquinoline-1,3(2H)dione (1c). White
					solid, yield: 85%. 1 H NMR (400 MHz, CDCl 3 ): d 8.68 (d, J = 7.2 Hz, 1H), 8.59
					(d, J = 7.6 Hz, 1H), 8.44 (d, J = 8.0 Hz, 1H), 8.06 (d, J = 8.0 Hz, 1H), 7.87
					(t, J = 7.6 Hz, 1H), 4.18 (t, J = 8.0 Hz, 2H), 1.78-1.71 (m, 2H), 1.47-1.29 (m,
					10H), 0.89 (t, J = 7.2 Hz, 3H); MS(ESI) calcd for C 20 H 23 BrNO 2 [M+H] +
					387.1, found: 387.1. 5.1.1.4.
					6-Bromo-2-dodecyl-1H-benzo[de]isoquinoline-1,3(2H)dione (1d). White solid,
					yield: 85%. 1 H NMR (400 MHz, CDCl 3 ): d 8.68 (d, J = 7.2 Hz, 1H), 8.59 (d, J =
					8.4 Hz, 1H), 8.44 (d, J = 7.6 Hz, 1H), 8.06 (d, J = 8.0 Hz, 1H), 7.87 (t, J =
					7.6 Hz, 1H), 4.18 (t, J = 7.6 Hz, 2H), 1.78-1.71 (m, 2H), 1.47-1.27 (m, 18H),
					0.90 (t, J = 7.2 Hz, 3H); MS(ESI) calcd for C 24 H 31 BrNO 2 [M+H] + 443.1,
					found: 443.2.</figDesc></figure>
```

Not really sure whats causing this, thought it would be useful to report it. I've attached the PDF that produced the above issues.

[1-s2.0-S0968089615005787-main.pdf](https://github.com/user-attachments/files/18877483/1-s2.0-S0968089615005787-main.pdf)

**Comment by haydn-jones** [comment]
When using the new light flavor option, it seems like things are generally handled better, but there are still various issues with things getting detected as references. If assistance would be helpful in resolving this, I'd be happy to help if I can.

**Comment by lfoppiano** [comment]
Hi @haydn-jones! Thanks for opening this issue. Indeed, there are a few issue with the chemical formulas. I'm not sure how many articles of similar subjects are there, probably a few, or none. Adding a few examples could be useful to get a solid improvement, at first. 

The first step is to find examples that are licenced as CC-BY or CC-0, then we could generate pre-annotated training data that needs to be corrected. The process may require some iterations, especially for annotating data for the fulltext model. I've described it [here](https://github.com/kermitt2/grobid/issues/1100#issuecomment-2078846116).

**Comment by haydn-jones** [comment]
Ok, so the idea is find some CC-BY/0 articles, and make some training data for the models? I'd be happy to do that, I assume the annotation guidelines section of the docs is where I should look to for guidance? I'm sure I can find a good number of papers in the PubMedCentral OA subset, how many do you think I should annotate?

**Comment by lfoppiano** [comment]
Yes, ideally I would start with 5-10 articles, possibly with a very disastrous output, and considering picking them from different publishers, if possible.  
I'm not sure is possible to get any elsevier's articles that are CC-BY (be careful at the licence, Elseiver is quite nasty as they publish many open access articles as CC-BY-ND, which won't allow redistribution of training data). 

and yes, the idea is to correct generated TEI files from Grobid. 

There is an extensive documentation, [here](https://grobid.readthedocs.io/en/latest/training/General-principles/), and before getting to correct the fulltext, we need to pass through [the segmentation model](https://grobid.readthedocs.io/en/latest/training/segmentation/) to make sure all text is extracted (the references being mixed up will be corrected in this model, BTW). Then we can take care of the [fulltext model](https://grobid.readthedocs.io/en/latest/training/fulltext/).

**Comment by haydn-jones** [comment]
Sounds good, 10 doesn't sound unreasonable. I'll start with one and send over what I've done to make sure I'm doing it properly.

**Comment by haydn-jones** [comment]
Looked at the output for this article https://pmc.ncbi.nlm.nih.gov/articles/PMC4279058/ (which I believe we should be able to use).

My understand is that we can *move*/*change*/*add*/*remove* the XML tags, but we cannot actually change the underlying text. With that, I tried to correct the segmentation output, can you let me know if these corrections are appropriate? I'll attach the old and new xml, along with the PDF. I inserted a few footnotes that it accidentally pulled into the body/bibl, though it seems to me they are actually headnotes, but grobid caught most of them as footnotes so I kept it like that for consistency. Is that also a mistake with the segmentation model?

I should be able to get 10 papers with distinct layouts / publishers, I have about 19,000 candidates right now in the PMC open access subset just need to dig through them.

[PMC4279058.pdf](https://github.com/user-attachments/files/18933185/PMC4279058.pdf)
[xml.tar.gz](https://github.com/user-attachments/files/18933210/xml.tar.gz)

**Comment by lfoppiano** [comment]
@haydn-jones indeed they are headnotes. The rest of the document looks good. 

Indeed the workflow requires the segmentation to work well in order to get to the fulltext. For other models this is not the case, but the way features are generated for fulltext require this workflow. 

If you correct more files, I can validate them.

**Comment by haydn-jones** [comment]
Finding 10 different paper formats / publishers turned out to be quite hard, but for most of these formats I have quite a few other pdfs we could use if necessary. Some of these did not have errors at the segmentation level, but all had errors somewhere in the pipeline. I think the file is too big to upload here, so I put it in google drive. https://drive.google.com/drive/folders/1Eoh5GpSJjAVTbVgaTnrueTK647oEuy_W?usp=drive_link

**Comment by lfoppiano** [comment]
Dear @haydn-jones, sorry took me quite some time to get to your pending task. I've revised the annotations, in general they were good, I have two comments: 
 - the first page's `page number`/`headnote`/`footnotes` should be annotated as `<front>` as they may be used for extracting bibliographic information from the header (Question in edit 2?). See in https://grobid.readthedocs.io/en/latest/training/segmentation/#header-front
 - the MDPI article tend to hide a lot of crap in white text that is normally not visible, in the first MDPI article there were duplicated headnote and page number, I remove the `<note` annotation as they should not be extracted 

To answer your questions, the article `PMC4402585` contains some unannotated numbers `123</lb>`. This are correct like that, it's the `crap` hidden under the Springer's logo 😄 

And, yes, we will include this files into the official training data

I'm attaching the validated files: [3_validated.zip](https://github.com/user-attachments/files/19269379/3_validated.zip)

**Comment by haydn-jones** [comment]
@lfoppiano Sorry I disappeared for a while, had an important deadline I really needed to crunch for. Now have some time to work on this again. I've expanded the set of papers a bit to include more publishers and verified that each paper is CC-BY or less restrictive. I've got segmentation more or less squared away, and I'm getting better results on that part (sometimes there are still errors on held out papers, but generally much better). 

I'm currently making progress on the fulltext training data, but I've got a question that I should ask about before I get all the way through. In a number of papers, there are "headings" that look like this:

![Image](https://github.com/user-attachments/assets/07975b68-67c8-4aa5-9d27-688ab2850114)

I can see why one would argue that this is *not* a header, but its also reasonable to consider it one. For GROBID, should this be treated as a heading or part of a paragraph?

Other than that, I think I should have everything finished and double checked this week!

**Comment by lfoppiano** [comment]
@haydn-jones Thanks, feel free to share the training data, I'd be happy to review them.

**Comment by haydn-jones** [comment]
@lfoppiano Took a bit longer than I wanted, but here are the annotations: https://drive.google.com/drive/folders/1YEE2R3BnJ5H59RPETAAG43duAINh5T_H?usp=drive_link

I think they are correct for the most part, but I may have handled headers incorrectly in some areas. For example, `PMC3159312` has paragraph headers that look like 

![Image](https://github.com/user-attachments/assets/c8906778-464a-4e58-ad82-d4e2c23379e2)

Not sure if these should be treated as actual headers or part of the paragraph.

**Comment by lfoppiano** [comment]
@haydn-jones thanks! I'm going to check the files next week. I should come up with an idea of what's best for annotating those titles 😄 Thanks a lot for the annotations 😃

**Comment by haydn-jones** [comment]
Sounds good, thanks!

**Comment by lfoppiano** [comment]
@haydn-jones could you create a PR with the new files? 
Please make sure you create it on a branch on your side and that I can push (tick "allow maintainers to access" or something like that when you create the PR). 
I will push updates if necessary directly in the PR, so that you can see the differences

**Comment by lfoppiano** [comment]
@kermitt2 does it makes sense to annotate all the different titles and subtitles as below? 

<img width="393" alt="Image" src="https://github.com/user-attachments/assets/526dde3b-b0e9-4268-9570-83706691657e" />

as follow: 

```xml
<head>■ EXPERIMENTAL SECTION<lb/></head>

<head>General Information.</head>

<p>All starting materials and reagents were<lb/> purchased from commercially available sources and used without<lb/> further purification except for CH 2 Cl 2 and benzene, which were<lb/> purified on a solvent purification system prior to the reaction. 1 [...]
type="biblio">39</ref> alkyne 13, cyclo-<lb/>adducts 15 and 16, and methoxytropones 3, 4 and iso-4 were<lb/> synthesized as previously reported. <ref type="biblio">14<lb/></ref> </p>

<head>DFT Calculations.</head>
<head>General Information for DFT Calculations.<lb/></head>

<p>Computational modeling was carried out using Schrodinger Suite&apos;s<lb/> Jaguar software. Unless otherwise noted, all calculations were<lb/> performed using the M06-2X functional and the 6-311G** basis<lb/> set in the gas phase at room temperature (298.15 K). This functional<lb/> and basis set was chosen for its cited accuracy for both conformational<lb/> and energetic accuracy. <ref type="biblio">40<lb/></ref> </p>
```

I saw a few examples only with head and subHead, but in this article the hierarchy is more deep. Maybe we should just limit to head and sub-head only? 


Article: https://pubs.acs.org/action/showCitFormats?doi=10.1021/acs.joc.5c00284&ref=pdf

**Comment by kermitt2** [comment]
Hi Luca, I think the current principle for training is to label indeed all possible heading and sub-heading just with `<head>`, independently from the hierarchical level. Then, recreating a hierarchy is a task left to the step of TEI reconstruction, but without good solution so far (I tried various simple things in the past, like clustering heading level based on the font style/size, but nothing was reliable at scale).

**Comment by haydn-jones** [comment]
@lfoppiano made the pull request, just realized that it contains updated segmentation and fulltext models. AFAIK there is an extended fulltext corpus, so while I can't train that model I can certainly update the segmentation model once we finalize the data.

I'd be happy to do another pass and correct all of the headings so we include the "paragraph" headings. At least that's how I've interpreted kermitt2's message (tag them as headers). I can also do a pass to make sure I'm splitting all of the headers (I may have forgot to split headings like `<head>2. Foo </lb>2.1 Bar </lb></head>`).

Edit: Maybe I should open another PR without the updated models as it is causing GitHub to not render diffs?

Edit 2: Did that.

**Comment by lfoppiano** [comment]
Thanks @haydn-jones. I've pushed the revised segmentation and 2 fulltexts. Overall the segmentation were good, I found only a few errors in the pages, that are quite tedious to check. Great work! 😄 
The fulltext files are generally harder. I did a few more corrections in particular there were a few missing references, and I've started tagging all `<head>` as @kermitt2 indicated (Thanks!! 👍). 

I also noticed that `PMC7248761.training.segmentation.tei.xml` was removed from the set, I was wondering why 

I let you do a second passage for adding the `<head>` in the fulltext files and then i can do a final validation

**Comment by haydn-jones** [comment]
@lfoppiano So with PMC7248761, there turned out to be a *ton* of hidden text in the paper for some reason, like entire paragraphs and figures are duplicated, which I didn't catch while doing the segmentation data. I removed it as I wasn't about to identify which parts of the text were visible / invisible. I'd be happy to give it another shot if we think it would be worthwhile.

[PMC7248761.pdf](https://github.com/user-attachments/files/20646706/PMC7248761.pdf)

![Image](https://github.com/user-attachments/assets/54358ee7-b951-42e4-98d6-e509a9d88a8d)

**Comment by lfoppiano** [comment]
@haydn-jones sorry, I forgot to answer that. Let's forget about it for now.

**Comment by lfoppiano** [comment]
@haydn-jones I've merged your PR. 🎉 Next I'll work on #1317 to try to find a way to have a conservative and robust approach for identifying at least the main sections of the article.
